src/hooks/useIntersectionObserver.ts

```diff
@@ -1,18 +1,20 @@
-import { useEffect, useRef, useState } from 'react'
+import { useLayoutEffect, useRef, useState } from 'react'
 
 const useIntersectionObserver = () => {
   const [observerRefElement, setObserverRefElement] = useState(null)
   const observerRef = useRef((element) => setObserverRefElement(element))
   const intersectionObserverRef = useRef<IntersectionObserver>(null)
   const [isIntersecting, setIsIntersecting] = useState(false)
 
-  useEffect(() => {
-    if (!intersectionObserverRef.current && observerRefElement) {
+  useLayoutEffect(() => {
+    const isSupported = typeof window === 'object' && window.IntersectionObserver
+
+    if (isSupported && !intersectionObserverRef.current && observerRefElement) {
```

Comment:
if not isSupported we should just say it intersecting?

Reply:
Hmm. I think it would be fine to do that. Will add it